website/server/models/newsPost.js

```diff
@@ -0,0 +1,90 @@
+import mongoose from 'mongoose';
+import baseModel from '../libs/baseModel';
+
+const { Schema } = mongoose;
+const POSTS_PER_PAGE = 10;
+
+export const schema = new Schema({
+  title: { $type: String },
+  author: { $type: String },
+  credits: { $type: String },
+  publishDate: { $type: Date },
+  published: { $type: Boolean },
+  text: { $type: String },
+}, {
+  strict: true,
+  minimize: false, // So empty objects are returned
+  typeKey: '$type', // So that we can use fields named `type`
+});
+
+schema.plugin(baseModel, {
+  noSet: ['_id'],
+  timestamps: true,
+});
+
+schema.statics.getLastPost = async function getLastPost () {
+  const post = await this.findOne({
+    published: true,
+    publishDate: { $lte: new Date() },
+  }).sort({ publishDate: -1 }).exec();
+  return post;
+};
+
+schema.statics.getNews = async function getNews (isAdmin, options = { page: 0 }) {
+  let query;
+  if (!isAdmin) {
+    query = this.find({
+      published: true,
+      publishDate: { $lte: new Date() },
+    })
+      .select('title publishDate credits text');
+  } else {
+    query = this.find();
+  }
+  let page = 0;
+  if (typeof options.page !== 'undefined') {
+    page = options.page;
+  }
+
+  return query.sort({ publishDate: -1 })
+    .limit(POSTS_PER_PAGE)
+    .skip(POSTS_PER_PAGE * Number(page))
+    .exec();
+};
+
+let cachedLastNewsPostID = null;
+let cachedLastNewsPostDate = null;
+let cachedLastNewsPostTitle = null;
+
+schema.statics.lastNewsPostID = async function lastNewsPostID () {
```

Comment:
I think this one and `lastNewsPostTitle` should be merged into a single function since in both cases both the `id` and `title` are fetched

Reply:
I changed the logic for this a bit

Reply:
I'd say this is a good solution, just two things:

- I'd stil merge `lastNewsPostID()` and `lastNewsPostTitle()` into a single function that gets and cache the last post
- When comparing dates I'd use moment's diff, it's a bit cleaner and `new Date()` are objects not numbers so it works because it gets casted to a number but it's a bit confusing, something like `moment().diff(timeStampCachedLastNews, 'minutes') < 5`